File: ZOJ/zoj3706.py
import sys

def num_of_mass(a,b,c):
	set_mass=set()
	for i in range(3):
		lefta=0
		righta=0
		if i==1:lefta+=a
		elif i==2:righta+=a
		for j in range(3):
			leftb=lefta
			rightb=righta
			if j==1:leftb+=b
			elif j==2:rightb+=b
			for k in range(3):
				leftc=leftb
				rightc=rightb
				if k==1:leftc+=c
				elif k==2:rightc+=c
				mass=int(abs(rightc-leftc))
				if mass>0 and mass not in set_mass:
					set_mass.add(mass)
	#print 'PYTHON: ' ,a,b,c,len(set_mass)
	return len(set_mass)

#fin=open('zoj3706_data_2')
#fout=open('out_py.txt','w')
#sys.stdin=fin
T=int(sys.stdin.readline())
COUNT=0
for n in range(T):
	line=sys.stdin.readline()
	a,b=map(int,line.split())
	tmpmax=0
	for i in range(1,a):
		tmpmax=max(tmpmax,num_of_mass(i,a-i,b))
	for i in range(1,b):
		tmpmax=max(tmpmax,num_of_mass(i,b-i,a))
	print tmpmax
	#print COUNT
	#COUNT+=1

File: ZOJ/zoj3706.cpp
#include <cstdio>
#include <cstdlib>
#include <iostream>
#include <set>

using namespace std;
int num_of_mass(int a, int b, int c){
	set<int> myset;
	for(int i=0;i<3;++i){
		int lefta=0;
		int righta=0;
		if(i==1)lefta+=a;
		else if(i==2)righta+=a;
		for(int j=0;j<3;++j){
			int leftb=lefta;
			int rightb=righta;
			if(j==1)leftb+=b;
			else if(j==2)rightb+=b;
			for(int k=0;k<3;++k){
				int leftc=leftb;
				int rightc=rightb;
				if(k==1)leftc+=c;
				else if(k==2)rightc+=c;
				int mass=abs(rightc-leftc);
				if(mass>0)myset.insert(mass);
			}
		}
	}
	//printf("CPLUSPLUS: %d %d %d %d\n",a,b,c,myset.size());
	return myset.size();
}
using namespace std;
int main(){
	//freopen("zoj3706_data","r",stdin);
	//freopen("out_cpp.txt","w",stdout);
	int T;
	scanf("%d",&T);
	for(int i=0;i<T;++i){
		int a,b;
		scanf("%d%d",&a,&b);
		int tmpmax=0;
		for(int j=1;j<a;++j){
			tmpmax=max(tmpmax,num_of_mass(j,a-j,b));
		}
		for(int j=1;j<b;++j){
			tmpmax=max(tmpmax,num_of_mass(j,b-j,a));
		}
		printf("%d\n",tmpmax);
	}
}
